### PR TITLE
STEK share plugin using Raft

### PR DESCRIPTION
**Pull request by duke8253**
This is a plugin used to share STEK among a cluster of ATS instances, it replaces the current session reuse plugin. This new plugin uses the Raft consensus algorithm (NuRaft library by Ebay https://github.com/eBay/NuRaft) for leader election.

The reason for creating this plugin is:
- The current session reuse plugin needs redis to run, and the leader election mechanism is somewhat messy.
- There is less and less need for sharing Session ID's, on our hosts it session reuse based on ID's are as low as 0.2%.

Have been testing this in our production hosts for a few weeks now, everything runs great other than the weird #8734 problem we were seeing.

### CODE REVIEW

**Comment by bryancall** [comment]
@moonchen is going to take a look at this

**Review by moonchen — CHANGES_REQUESTED** [review]

File: plugins/experimental/stek_share/log_store.cc [hunk 1]
```diff
@@ -0,0 +1,263 @@
+/************************************************************************
+Copyright 2017-2019 eBay Inc.
+Author/Developer(s): Jung-Sang Ahn
```
[the remark this answers is not in the document]
Comment (duke8253) [reply plugins/experimental/stek_share/log_store.cc:3]: I would assume so since it's their file.

**Review by moonchen — APPROVED** [review]

File: plugins/experimental/stek_share/common.h [hunk 2]
```diff
@@ -0,0 +1,72 @@
+/** @file
+
+  commmon.h - Things that need to be everywhere
```
Comment (randall) [inline plugins/experimental/stek_share/common.h:3]: spelling

File: plugins/experimental/stek_share/stek_share.cc [hunk 3]
```diff
@@ -0,0 +1,446 @@
...
+        } else {
+          stek_share_server.last_updated_ = time(nullptr);
+          TSDebug(PLUGIN, "Update SSL Ticket Key succeeded.");
+        }
+      }
+    }
+    // print_status();
+
+    // Wakeup every 10 seconds to check whether there is a new key to use.
+    // We do this because if a server is lagging behind, either by loosing connection or joining late,
```
Comment (randall) [inline plugins/experimental/stek_share/stek_share.cc:412]: spelling (losing)

File: plugins/experimental/stek_share/stek_share.cc [hunk 4]
```diff
@@ -0,0 +1,446 @@
...
+        TSDebug(PLUGIN, "Updating SSL Ticket Key...");
+        if (TSSslTicketKeyUpdate(reinterpret_cast<char *>(stek_share_server.ticket_keys_), SSL_TICKET_KEY_SIZE * 2) == TS_ERROR) {
+          TSDebug(PLUGIN, "Update SSL Ticket Key failed.");
+        } else {
+          stek_share_server.last_updated_ = time(nullptr);
+          TSDebug(PLUGIN, "Update SSL Ticket Key succeeded.");
+        }
+      }
+    }
+    // print_status();
```
Comment (randall) [inline plugins/experimental/stek_share/stek_share.cc:409]: remove commented out code.

File: plugins/experimental/stek_share/common.h (hunk 2 above)
Comment (duke8253) [reply plugins/experimental/stek_share/common.h:3]: which part?

Comment (randall) [reply plugins/experimental/stek_share/common.h:3]: Common not commmon

**Review by randall — APPROVED** [review]

**Comment by zwoop** [comment]
Cherry-picked to v9.2.x

**Pull request merged**